added color change fix for CodeFolding arrows and change of cursor

geekodour opened the pull request:
The hover color change has been fixed, it's highlighted when hovered and dimmed when not.
Also, added the cursor to be a pointer when over the arrows line. 
Issue on thimble : https://github.com/mozilla/thimble.mozilla.org/issues/1874#issuecomment-286223341
I've uploaded the working of it [here(gif).](http://i.imgur.com/r0eEe3n.gifv)

Fixes https://github.com/mozilla/thimble.mozilla.org/issues/1874

Pomax commented:
Hi @geekodour, instead of saying "Issue #1874 from Thimble" in the title, can you take that part out, and then in the PR comment itself note "fixes <link to issue on thimble>"? This makes github automatically set up crosslinks between the PR and the issue, making PR/issue administration easier.

geekodour commented:
@Pomax I've removed "Issue #1874 from Thimble" as you stated, but pushing another change with a new commit message will probably not work with "fix" because thimble is another repo and I am pushing change on the brackets repository. Am I right? In addition to that, I've added the link to the issue on thimble in the main comment.

Pomax commented:
As long as it's in the main comment, github will maintain the crosslink, but that "fixes ..." was not a suggestion: it needs to *literally* be `fixes ......` in order for github to set up the crosslink, so not "issue on thimble: ..." =)

(the `fixes #12345` is github.com-specific syntax that makes their backend setup crosslinks, and automatically close PR-associated issues if they're in the same repo)

flukeout commented:
Testing it out now!

flukeout commented:
@Pomax Except just using #1234 wouldn't work because that would reference issue 1234 in _this_ repo? Does it work with a full URL? I've added it to the message so we'll test it.

As far as the PR, thanks very much for submitting it. In reviewing it I've noticed a few things

* The hover effect happens no matter where you hover in the gutter, even in places when you can't click the arrows
* The cursor changes to a pointer along the entire height of the arrow div, which is incorrect because it is only useful to click it directly overtop of the arrow.

I think this is just because of the design of this feature. There's an element that overlaps all of the arrows so we can't really apply :hover styles to them it seems.

Because of this could we instead:

* Remove ALL hover and cursor effects from the arrows?
* And just make sure they are always the same color when hovered or not.

Thank you!

geekodour commented:
okay great, I'll send the changes very soon. thanks for reviewing.

geekodour commented:
There could be another possibility for achieving this, changing these lines where the over-gutter class is assigned, here: https://github.com/mozilla/brackets/blob/master/src/extensions/default/CodeFolding/main.js#L282-L297

humphd commented:
@geekodour a note to let you know that @flukeout is assigned to review this, but has limited time this week.  So he may be a few days before he can get to it. Just so you didn't feel like it was being ignored.

flukeout commented on src/extensions/default/CodeFolding/main.less:
```diff
@@ -63,11 +57,11 @@
         vertical-align: top;
         line-height: 0.8em;
     }
-}
```
Can you add back the line break at the end of this file?

geekodour replied:
thankyou for reviewing. I've made the changes.

geekodour replied:
and I actually rm'ed the main.css file, did you mean not to modify main.css at all (by remove from PR)?

geekodour replied:
i messed things up in the latest commit, please check back in some time.

geekodour replied:
now, it's okay. the PR has changes to main.less only.

geekodour replied:
let me know if I have to rollback everything and send it again, I got confused because I was using hg to send some patch to firefox, the workflows are similar yet different so I got confused. sorry. 😢

Pomax replied:
If you just need to roll back a single file: worry not! If you've accidentally deleted your files, you can get it back from any commit that had it previously. So if you're in the directory with main.css, and you do:
```
$> git checkout main.css master
```
you will get the `main.css` file back, as it existed in the master branch. Or, if you need it from commit hash 8236b721693df4h1263479f8d162:
```
$> git checkout main.css 8236b721693df4h1263479f8d162
```

flukeout commented on src/extensions/default/CodeFolding/main.css:
```diff
@@ -1,2 +1 @@
-/* XXXBramble - created from main.less with lessc --clean-css */
-.CodeMirror-foldgutter-open:after{content:"\25bc";font-size:1.1em;color:#ccc}.CodeMirror-foldgutter-folded:after{content:"\25b6";font-size:1.1em;color:#999}.CodeMirror-activeline .CodeMirror-foldgutter-open:after,.CodeMirror.over-gutter .CodeMirror-foldgutter-open:after{color:#aaa}.CodeMirror-foldgutter,.CodeMirror-foldgutter-blank,.CodeMirror-foldgutter-folded,.CodeMirror-foldgutter-open{width:1 ...
\ No newline at end of file
+.CodeMirror-foldgutter-open:after{content:"\25bc";font-size:1.1em;color:#999}.CodeMirror-foldgutter-folded:after{content:"\25b6";font-size:1.1em;color:#999}.CodeMirror-foldgutter,.CodeMirror-foldgutter-blank,.CodeMirror-foldgutter-folded,.CodeMirror-foldgutter-open{width:1.1em;cursor:default;line-height:100%}.CodeMirror-foldgutter-folded,.CodeMirror-foldgutter-open{padding-top:2px}.CodeMirror-gut ...
```
Can you remove this file from the PR entirely?

flukeout commented:
@geekodour - Hey this is great - just tested it and I think it's good to go. I left some comments up above and once those are fixed - we can land this. Thank you!

humphd commented:
@geekodour it looks like we're going in a different direction, see https://github.com/mozilla/brackets/pull/667.  I'm going to close this, since it won't be necessary with @flukeout's change.  Thank you for working on this, and feel free to keep working with us on another bug.  It's been great having you contribute and be part of the project.
